src/test/java/uk/gov/justice/digital/delius/service/SentenceServiceTest.java

```diff
@@ -0,0 +1,154 @@
+        Optional<CustodialStatus> status = sentenceService.getCustodialStatus(CRN, CONVICTION_ID, SENTENCE_ID);
+
+        assertThat(status).isEmpty();
+    }
+
+    @Test
+    public void whenConvictionDoesNotMatchSentence_thenReturnEmpty() {
+        when(offenderRepository.findByCrn(CRN)).thenReturn(Optional.of(offender));
+        when(disposalRepository.findByDisposalId(SENTENCE_ID)).thenReturn(Optional.of(disposal));
+        when(offender.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getEvent()).thenReturn(Event.builder()
+                .eventId(9999999L)
+                .build());
+        Optional<CustodialStatus> status = sentenceService.getCustodialStatus(CRN, CONVICTION_ID, SENTENCE_ID);
+
+        assertThat(status).isEmpty();
+    }
+
+    @Test
+    public void whenSentenceIsSoftDeleted_thenReturnEmpty() {
+        when(offenderRepository.findByCrn(CRN)).thenReturn(Optional.of(offender));
+        when(disposalRepository.findByDisposalId(SENTENCE_ID)).thenReturn(Optional.of(disposal));
+        when(offender.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getEvent()).thenReturn(event);
+        when(disposal.isSoftDeleted()).thenReturn(true);
+        Optional<CustodialStatus> status = sentenceService.getCustodialStatus(CRN, CONVICTION_ID, SENTENCE_ID);
+
+        assertThat(status).isEmpty();
+    }
+
+    @Test
+    public void whenCustodyIsNull_thenReturnEmpty() {
+        when(offenderRepository.findByCrn(CRN)).thenReturn(Optional.of(offender));
+        when(disposalRepository.findByDisposalId(SENTENCE_ID)).thenReturn(Optional.of(disposal));
+        when(offender.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getEvent()).thenReturn(event);
+        when(disposal.isSoftDeleted()).thenReturn(false);
+        when(disposal.getCustody()).thenReturn(null);
+        Optional<CustodialStatus> status = sentenceService.getCustodialStatus(CRN, CONVICTION_ID, SENTENCE_ID);
+
+        assertThat(status).isEmpty();
+    }
+
+    @Test
+    public void whenCustodyIsSoftDeleted_thenReturnEmpty() {
+        when(offenderRepository.findByCrn(CRN)).thenReturn(Optional.of(offender));
+        when(disposalRepository.findByDisposalId(SENTENCE_ID)).thenReturn(Optional.of(disposal));
+        when(offender.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getOffenderId()).thenReturn(OFFENDER_ID);
+        when(disposal.getEvent()).thenReturn(event);
+        when(disposal.isSoftDeleted()).thenReturn(false);
+        when(disposal.getCustody()).thenReturn(custody);
+        when(custody.isSoftDeleted()).thenReturn(true);
+        Optional<CustodialStatus> status = sentenceService.getCustodialStatus(CRN, CONVICTION_ID, SENTENCE_ID);
```

Comment:
I rarely use the approach of mocking entities, but it's actually quite nice in that the only data mentioned is the data needed for the test - no test entity full of irrelevant data to reason about. :+1:

Reply:
Personally I prefer starting from EntityBuilder.anOffender().toBuilder() etc and manipulating, but both work!